module/dkg/client.go

```diff
@@ -0,0 +1,246 @@
+package dkg
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/onflow/cadence"
+
+	"github.com/onflow/flow-core-contracts/lib/go/templates"
+
+	sdk "github.com/onflow/flow-go-sdk"
+	sdkcrypto "github.com/onflow/flow-go-sdk/crypto"
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/model/flow"
+	model "github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/module"
+)
+
+// Client is a client to the Flow DKG contract. Allows functionality to Broadcast,
+// read a Broadcast and submit the final result of the DKG protocol
+type Client struct {
+	dkgContractAddress string
+	accountAddress     sdk.Address
+	accountKeyIndex    uint
+	flowClient         module.SDKClientWrapper
+	signer             sdkcrypto.Signer
+
+	env templates.Environment // the required contract addresses to be used by the Flow SDK
+}
+
+// NewClient initializes a new client to the Flow DKG contract
+func NewClient(flowClient module.SDKClientWrapper, signer sdkcrypto.Signer, dkgContractAddress, accountAddress string, accountKeyIndex uint) *Client {
+	return &Client{
+		flowClient:         flowClient,
+		dkgContractAddress: dkgContractAddress,
+		signer:             signer,
+		accountKeyIndex:    accountKeyIndex,
+		accountAddress:     sdk.HexToAddress(accountAddress),
+		env:                templates.Environment{DkgAddress: dkgContractAddress},
+	}
+}
+
+// Broadcast broadcasts a message to all other nodes participating in the
+// DKG. The message is broadcast by submitting a transaction to the DKG
+// smart contract. An error is returned if the transaction has failed has
+// failed
+func (c *Client) Broadcast(msg model.DKGMessage) error {
+
+	ctx := context.Background()
```

Comment:
One other enhancement I think would be worth including is to improve visibility is include some logging in these methods, so we would be able to see if we get stuck in a loop in one of these methods for longer than expected.
We could log:
* when submitting transaction `SubmitTransaction`
* when checking transaction each time `GetTransactionResult` with:
  * time since we started (ie. for this method, time since `Broadcast` was invoked)
  * number of retries

In the happy path case we only send one message in both QC and DKG, so I think it's OK to log these at `info` level

Reply:
@danuio Were you thinking to add the logging change in a separate PR? Not seeing any logging added in the second review.

Reply:
Yes, there should be PR out for it today.